### ISSUE TITLE
[FEATURE REQUEST] Allow to disable Open Distro audit logs

### ISSUE THREAD

**Issue by seriva** [epiphany-platform/epiphany#3060]
**Is your spike related to a problem or idea? Please describe.**
By default ELK stack doesn't have any policy management for its default indices. There are security indices that are producing data all the time which ends with taking over all possible shards on ElasticSearch and ELK stack stops working.

Example issue from a customer:

`
{"agent":{"ephemeral_id":"cfcd68d5-ec76-424e-b2eb-1e89d1ac57c3","hostname":"lahcmonitor01d","id":"474b78b4-60ff-4fc5-9147-acce120f1dd2","name":"lahcmonitor01d","type":"filebeat","version":"7.9.2"},"ecs":{"version":"1.5.0"},"host":{"name":"lahcmonitor01d"},"input":{"type":"log"},"log":{"file":{"path":"/var/log/messages"},"offset":703865869},"message":"Feb 21 09:05:19 lahcmonitor01d performance-analyzer-agent-cli: 09:05:19.140 [pa-reader] ERROR com.amazon.opendistro.elasticsearch.performanceanalyzer.reader.ReaderMetricsProcessor - Error reading file '/usr/share/elasticsearch/data/batch_metrics_enabled.conf': java.nio.file.NoSuchFileException: /usr/share/elasticsearch/data/batch_metrics_enabled.conf"}, Private:file.State{Id:"native::3195-64773", PrevId:"", Finished:false, Fileinfo:(*os.fileStat)(0xc000c24680), Source:"/var/log/messages", Offset:703866222, Timestamp:time.Time{wall:0xc07d018e2a7d621c, ext:8371919972062144, loc:(*time.Location)(0x419f7a0)}, TTL:-1, Type:"log", Meta:map[string]string(nil), FileStateOS:file.StateOS{Inode:0xc7b, Device:0xfd05}, IdentifierName:"native"}, TimeSeries:false}, Flags:0x1, Cache:publisher.EventCache{m:common.MapStr(nil)}} (status=400): {"type":"validation_exception","reason":"Validation Failed: 1: this action would add [6] total shards, but this cluster currently has [999]/[1000] maximum shards open;"}
`

This indices are created by default (tbh same as filebeat*) so there should be some default policy to prevent this situation from happening. 

**Describe the outcome you'd like**
Check what kind of Policy management we can offer to keep this issue from happening in the form of additional issues. Note that these (might/should) need to cover the following:

- Elasticsearch for version 1.0.3

See comments below for approach taken.

**What is the reason or source for the spike**
Internal team

**Additional context**
None

---

**DoD checklist**

- Changelog
  - [x] updated
  - [ ] not needed
- COMPONENTS.md
  - [ ] updated
  - [x] not needed
- Schema
  - [x] updated
  - [ ] not needed
- Backport tasks
  - [x] created
  - [ ] not needed
- Documentation
  - [ ] added
  - [x] updated
  - [ ] not needed
- [ ] Feature has automated tests
- [x] Automated tests passed (QA pipelines)
  - [x] apply
  - [ ] upgrade
  - [ ] backup/restore
- [ ] Idempotency tested
- [ ] All conversations in PR resolved
- [ ] Solution meets requirements and is done according to design doc
- [x] Usage compliant with license

**Comment by atsikham** [comment]
<img width="997" alt="image" src="https://user-images.githubusercontent.com/6271862/163997832-0ffc1d40-c217-4f4b-853d-3bab12e8119d.png">


https://github.com/epiphany-platform/epiphany/blob/develop/docs/home/howto/LOGGING.md#how-to-manage-opendistro-for-elasticsearch-data

**Comment by atsikham** [comment]
We have quite a good and short documentation with a policy example - https://github.com/epiphany-platform/epiphany/blob/develop/docs/home/howto/LOGGING.md#how-to-manage-opendistro-for-elasticsearch-data[](https://github.com/epiphany-platform/epiphany/blob/develop/docs/home/howto/LOGGING.md#how-to-manage-opendistro-for-elasticsearch-data)

In fact this policy should depend on the size of machines storage and their number, which is configurable. We could disable an audit logging by default, or make this option configurable:

- https://github.com/epiphany-platform/epiphany/blob/0f0e01a784afc00931116a13a1a638ba6a72e9d8/ansible/playbooks/roles/opendistro_for_elasticsearch/templates/elasticsearch.yml.j2#L135
- https://opendistro.github.io/for-elasticsearch-docs/docs/security/audit-logs/

Or configure some policy, but we can't be sure in the requirement `prevent this situation from happening` - this should be analysed by the user, and taking into account the disk size, based on their experience, configured.

Security/audit can be disabled/configured via Kibana UI, the same with policies:

<img width="1440" alt="image" src="https://user-images.githubusercontent.com/6271862/164425764-60971c3c-1949-4d23-8d81-bb0e732f81d1.png">

<img width="1440" alt="image" src="https://user-images.githubusercontent.com/6271862/164425868-7193a029-a7d5-4e70-89fa-bb9506824888.png">

**Comment by atsikham** [comment]
According to documentation it's almost the same for opensearch:

- https://opensearch.org/docs/latest/security-plugin/audit-logs/index/
- https://opensearch.org/docs/latest/im-plugin/ism/index/

As I see in PR #3093, we are going to enable audit logging by default again:
<img width="1114" alt="image" src="https://user-images.githubusercontent.com/6271862/164429231-53c0b5b8-8e2c-4802-a798-c57828f8b1dd.png">

**Comment by atsikham** [comment]
❗ ❗ ❗ Similar work should be done for v2.0.1 after Opensearch migration.

**Comment by seriva** [comment]
> ❗ ❗ ❗ Similar work should be done for v2.0.1 after Opensearch migration.

@rafzei Is this something we can include in the current OpenSearch PR maybe?

**Comment by rafzei** [comment]
This PR is quite big already. I would recommend doing it in the separate one.

**Comment by seriva** [comment]
> This PR is quite big already. I would recommend doing it in the separated one.

Could you create a new issue for that for 2.0.1, linking back to this issue and blocked by your opensearch one?

**Comment by seriva** [comment]
The OpenSearch part is deferred to #3140, we can close this once tested for OpenDistro.
